diff --git a/drivers/hv/dxgkrnl/dxgmodule.c b/drivers/hv/dxgkrnl/dxgmodule.c
index c2a4a2a21..0941ab70a 100644
--- a/drivers/hv/dxgkrnl/dxgmodule.c
+++ b/drivers/hv/dxgkrnl/dxgmodule.c
@@ -28,6 +28,18 @@
 #undef dev_fmt
 #define dev_fmt(fmt)   "dxgk: " fmt
 
+/*
+ * GPU paravirtualization global DXGK channel
+ * {DDE9CBC0-5060-4436-9448-EA1254A5D177}
+ */
+#define HV_GPUP_DXGK_GLOBAL_GUID .guid = GUID_INIT(0xdde9cbc0, 0x5060, 0x4436, 0x94, 0x48, 0xea, 0x12, 0x54, 0xa5, 0xd1, 0x77)
+
+/*
+ * GPU paravirtualization per virtual GPU DXGK channel
+ * {6E382D18-3336-4F4B-ACC4-2B7703D4DF4A}
+ */
+#define HV_GPUP_DXGK_VGPU_GUID .guid = GUID_INIT(0x6e382d18, 0x3336, 0x4f4b, 0xac, 0xc4, 0x2b, 0x77, 0x3, 0xd4, 0xdf, 0x4a)
+
 /*
  * Interface from dxgglobal
  */
@@ -175,7 +187,7 @@ void signal_host_cpu_event(struct dxghostevent *eventhdr)
    }
    if (event->cpu_event) {
        DXG_TRACE("signal cpu event");
-       eventfd_signal(event->cpu_event, 1);
+       eventfd_signal(event->cpu_event);
        if (event->destroy_after_signal)
            eventfd_ctx_put(event->cpu_event);
    } else {
